app/models/form_submission_attempt.rb

```diff
@@ -66,10 +66,10 @@ def log_status_change
       to_state: aasm.to_state,
       event: aasm.current_event
     }
-    if failure?
+    if aasm.current_event == 'fail!'
```

Comment:
I was recently looking through this and wondering about if this was working. Thanks for cleaning it up!